core/shared/src/main/scala/zio/Fiber.scala

```diff
@@ -316,9 +290,17 @@ trait Fiber[+E, +A] { self =>
     ZManaged.make(UIO.succeed(self))(_.interrupt)
 
   /**
-   * The trace of the fiber. Currently only single runtime fibers have traces.
+   * Performs an effect depending on whether this is a runtime or synthetic
+   * fiber.
    */
-  def trace: UIO[Option[ZTrace]]
+  final def typeSpecific[R, E1 >: E, B](
```

Comment:
We could also call this `foldM`, e.g. after `Either#fold`, since `Fiber` is now a sum type of two possibilities.

Also you really only need `fold`, i.e. fold to a generic `Z`, which will be no less powerful than this one (since that `Z` could also be a ZIO effect).

Reply:
Yes you’re right about making it more generic. I thought about fold but was worried it could be confused with folding over the success and failure cases.